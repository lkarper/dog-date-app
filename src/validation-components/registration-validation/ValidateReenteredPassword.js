import React, { useEffect } from 'react';

const ValidateReenteredPassword = (props) => {
    const { reenteredPasswordError } = props;

    useEffect(() => {
        const { setReenteredPasswordError, password, reenteredPassword } = props;
        if (password === reenteredPassword) {
            setReenteredPasswordError(null);
        } else {
            setReenteredPasswordError('Passwords do not match.');
        }
    }, [props]);

    if (!reenteredPasswordError) {
        return <p id="reenter-password-validation">Passwords match.</p>;
    } else {
        return <p id="reenter-password-validation">{reenteredPasswordError}</p>; 
    }
}

export default ValidateReenteredPassword;