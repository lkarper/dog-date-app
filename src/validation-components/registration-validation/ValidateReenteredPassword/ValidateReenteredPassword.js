import React, { useEffect } from 'react';
import PropTypes from 'prop-types';

const ValidateReenteredPassword = (props) => {
    const { 
        password, 
        reenteredPassword, 
        reenteredPasswordError,
        setReenteredPasswordError,
    } = props;

    useEffect(() => {
        if (password === reenteredPassword) {
            setReenteredPasswordError('');
        } else {
            setReenteredPasswordError('Passwords do not match.');
        }
    }, [password, reenteredPassword, setReenteredPasswordError]);

    if (reenteredPasswordError) {
        return (
            <p 
                className='ValidateReenteredPassword__validator error'
                id="reenter-password-validation"
            >
                {reenteredPasswordError}
            </p> 
        );
    }

    return (
        <p 
            className='ValidateReenteredPassword__validator valid'
            id="reenter-password-validation"
        >
            Passwords match.
        </p>
    );
    
}

ValidateReenteredPassword.defaultProps = {
    password: '', 
    reenteredPassword: '', 
    reenteredPasswordError: '',
    setReenteredPasswordError: () => {},
};

ValidateReenteredPassword.propTypes = {
    password: PropTypes.string.isRequired, 
    reenteredPassword: PropTypes.string.isRequired, 
    reenteredPasswordError: PropTypes.string.isRequired,
    setReenteredPasswordError: PropTypes.func.isRequired,
};

export default ValidateReenteredPassword;
