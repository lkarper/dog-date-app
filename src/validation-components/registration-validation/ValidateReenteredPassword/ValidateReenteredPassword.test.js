import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ValidateReenteredPassword from './ValidateReenteredPassword';

describe('ValidateReenteredPassword component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<ValidateReenteredPassword />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<ValidateReenteredPassword />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
