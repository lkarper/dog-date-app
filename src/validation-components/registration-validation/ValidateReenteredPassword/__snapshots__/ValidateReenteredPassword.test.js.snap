// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValidateReenteredPassword component renders the UI as expected 1`] = `
<ValidateReenteredPassword
  password=""
  reenteredPassword=""
  reenteredPasswordError=""
  setReenteredPasswordError={[Function]}
>
  <p
    className="ValidateReenteredPassword__validator valid"
    id="reenter-password-validation"
  >
    Passwords match.
  </p>
</ValidateReenteredPassword>
`;
